src/Digdir.Domain.Dialogporten.WebApi/Common/Authentication/TokenIssuerCache.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using Microsoft.IdentityModel.Protocols;
+using System.Collections.ObjectModel;
+
+namespace Digdir.Domain.Dialogporten.WebApi.Common.Authentication;
+
+public interface ITokenIssuerCache
+{
+    public Task<string?> GetIssuerForScheme(string schemeName);
+}
+
+public class TokenIssuerCache : ITokenIssuerCache, IDisposable
+{
+    private readonly Dictionary<string, string> _issuerMappings = new();
+    private readonly SemaphoreSlim _initializationSemaphore = new(1, 1);
+    private bool _initialized;
+    private readonly IEnumerable<JwtBearerTokenSchemasOptions>? _jwtTokenSchemas;
```

Comment:
1. Du kan gå ut i fra at du har denne, så ta bort nullable. Dersom du ikke har det kan du egentlig kaste exceptions i ctor. 
2. Bruk `IReadOnlyCollection` i stede for `IEnumerable`.